packages/aws-appsync/src/client.js

```diff
@@ -43,18 +43,21 @@ class AWSAppSyncClient extends ApolloClient {
             res = resolve;
         });
 
-        const store = disableOffline ? null : createStore(
+        const store = (disableOffline || ownCache) ? null : createStore(
             this,
             () => {
                 store.dispatch({ type: 'REHYDRATE_STORE' });
                 res(this);
             },
             conflictResolver,
         );
-        const cache = disableOffline ? new InMemoryCache() : new OfflineCache(store);
+
+        const cache = disableOffline
+            ? ownCache ? owncache : new InMemoryCache()
```

Comment:
This assignment would be cleaner if you swap the ternary operators:
```
const cache = ownCache ? ownCache
    : disableOffline ? new InMemoryCache()
    : new OfflineCache(store);
```
There also seems to be a typo on line 56: `owncache` should be `ownCache`.

Reply:
Agreed with this feedback as it is a lot more readable.